hoomd/write/dcd.py

```diff
@@ -0,0 +1,104 @@
+        * dump.dcd will not write out data at time steps that already are present in the dcd file to maintain a
+          consistent timeline
+
+    Attributes:
+        filename (str): File name to write.
+        trigger (hoomd.trigger.Trigger): Select the timesteps to write.
+        filter (hoomd.filter.ParticleFilter): Select the particles to write.
+        overwrite (bool): When False, an existing DCD file will be appended to. When True, an existing DCD
+                          file *filename* will be overwritten.
+        unwrap_full (bool): When False, particle coordinates are always written inside the simulation box.
+                            When True, particles will be unwrapped into their current box image before writing to the dcd file.
+        unwrap_rigid (bool): When False, individual particles are written inside the simulation box which
+               breaks up rigid bodies near box boundaries. When True, particles belonging to the same rigid body will be
+               unwrapped so that the body is continuous. The center of mass of the body remains in the simulation box, but
+               some particles may be written just outside it. *unwrap_rigid* is ignored when *unwrap_full* is True.
+        angle_z (bool): When True, the particle orientation angle is written to the z component
+    """
+    def __init__(self,
+                 filename,
+                 trigger,
+                 filter=All(),
+                 overwrite=False,
+                 unwrap_full=False,
+                 unwrap_rigid=False,
+                 angle_z=False):
+
+        # initialize base class
+        super().__init__(trigger)
+        self._param_dict.update(
+            ParameterDict(filename=str(filename),
+                          filter=ParticleFilter,
+                          overwrite=bool(overwrite),
+                          write_settings=[bool],
+                          _defaults=dict(filter=filter,
+                                         write_settings=[unwrap_full,
+                                                         unwrap_rigid,
+                                                         angle_z])))
+
+    def _attach(self):
+        write_settings = []
+        if self.write_settings is not None:
+            write_settings = [setting for setting in self.write_settings]
+        group = self._simulation.state._get_group(self.filter)
+
+        self._cpp_obj = _hoomd.DCDDumpWriter(self._simulation.state._cpp_sys_def,
+                                             self.filename,
+                                             int(self.trigger.period),
+                                             group,
+                                             self.overwrite)
+        self._cpp_obj.setUnwrapFull(write_settings[0])
+        self._cpp_obj.setUnwrapRigid(write_settings[1])
+        self._cpp_obj.setAngleZ(write_settings[2])
```

Comment:
```suggestion
```
Once these are available as properties in the pybind11 export, `super.attach()` will set them.